custom/enikshay/case_utils.py

```diff
@@ -71,19 +71,43 @@ def get_open_episode_case_from_person(domain, person_case_id):
         raise ENikshayCaseNotFound(
             "Person with id: {} exists but has no open occurrence cases".format(person_case_id)
         )
-    occurrence_case = open_occurrence_cases[0]
-    episode_cases = case_accessor.get_reverse_indexed_cases([occurrence_case.case_id])
+    return open_occurrence_cases[0]
+
+
+def get_open_episode_case_from_occurrence(domain, occurrence_case_id):
+    """
+    Gets the first open 'episode' case for the person
+
+    Assumes the following case structure:
+    Occurrence <--ext-- Episode
+
+    """
+    case_accessor = CaseAccessors(domain)
+    episode_cases = case_accessor.get_reverse_indexed_cases([occurrence_case_id])
     open_episode_cases = [case for case in episode_cases
                           if not case.closed and case.type == CASE_TYPE_EPISODE and
                           case.dynamic_case_properties().get('episode_type') == "confirmed_tb"]
     if open_episode_cases:
         return open_episode_cases[0]
     else:
         raise ENikshayCaseNotFound(
-            "Person with id: {} exists but has no open episode cases".format(person_case_id)
+            "Occurrence with id: {} exists but has no open episode cases".format(occurrence_case_id)
         )
 
 
+def get_open_episode_case_from_person(domain, person_case_id):
+    """
+    Gets the first open 'occurrence' case for the person
+
+    Assumes the following case structure:
+    Person <--ext-- Occurrence <--ext-- Episode
+
+    """
+    return get_open_episode_case_from_occurrence(
+        get_open_occurrence_case_from_person(person_case_id).case_id
```

Comment:
this needs a `domain`.

There's a test [here](https://github.com/dimagi/commcare-hq/blob/f1ee1f44189b9371390becc98312cbd14959da39/custom/enikshay/tests/test_case_utils.py#L73-L74) you can run to validate this is working.